common/utils/filters.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  CatalogueAggregationBucket,
+  CatalogueAggregations,
+} from '@weco/common/model/catalogue';
+import { nodeListValueToArray, inputValue } from '@weco/common/utils/forms';
+import { LinkProps } from 'next/link';
+
+import {
+  worksLink,
+  imagesLink,
+} from '@weco/common/services/catalogue/ts_routes';
+
+export const getFilterByCount = (
+  bucketFilter: CatalogueAggregationBucket[]
+): CatalogueAggregationBucket[] => {
+  return bucketFilter.filter(items => {
+    return items.count > 0;
+  });
+};
+
+export const getAggregationFilterByName = (
+  aggregations: CatalogueAggregations | undefined,
+  name: string
+): CatalogueAggregationBucket[] | [] => {
+  return aggregations && aggregations?.[name]?.buckets
+    ? getFilterByCount(aggregations?.[name]?.buckets)
+    : [];
+};
+
+type RadioGroupOption = {
+  value: string;
+  id: string;
+  label: string;
+};
+
+// sort by the highest count
+export const sortByCount = (
+  a: CatalogueAggregationBucket,
+  b: CatalogueAggregationBucket
+): number => {
+  if (a.count > b.count) {
+    return -1;
+  } else {
+    return 1;
+  }
+};
+
+export const getAggregationRadioGroup = (
+  aggregation: CatalogueAggregationBucket[],
+  prefixId: string
+): RadioGroupOption[] => {
+  const radioOption = aggregation.sort(sortByCount).map(item => {
+    const id = item?.data?.label.replace(/\s/, '-').toLocaleLowerCase();
```

Comment:
This ID should come across from the API no?

Reply:
The id does not come through actually. :(

Reply:
Could you send me an example please?

Reply:
OK - this makes sense as the values from the API are labels, not IDs.

Reply:
https://api.wellcomecollection.org/catalogue/v2/works?aggregations=subjects&page=1&pageSize=10&query=test&subjects.label=AIDS Serodiagnosis.